src/App.jsx

```diff
@@ -50,12 +72,22 @@ function App() {
           return new Date(a.dueDate) - new Date(b.dueDate);
         }
       },
+
       {
         title: "Tags",
         dataIndex: "tags",
         key: "tags",
+        filters: tagsFilter,
+        onFilter: (value, todo) => {
+          for (let i = 0; i < todo.tags.length; i++) {
+            if (value === todo.tags[i]) {
+              return todo
+            }
+          }
```

Comment:
I don't understand why you use `Array.includes` at one line and then `for loop` at another for the same logic.
I suppose you thought you have to return the `todo` object. But...

Just like the callback function for `Array.filter`, the `onFilter` function should return `boolean`, therefore `true` or `false` depending whether or not the tag is included in the `todo.tags`. 

So, instead of `return todo` once the tag was found, it should return `true` or `false`.
I suppose the code is working because the `todo` object is evaluated to [truthy value](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) and there is no case for when the tag was not found.
That can be explained as when a function does not have a return statement it returns `undefined` by default
which is evaluated to [falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).

This is how it should be written correctly.
```
for (let i = 0; i < todo.tags.length; i++) {
    if (value === todo.tags[i]) {
        return true;
    }
}
return false;
```
Or you could take advantage of the `includes` method as per the suggestion.
```suggestion
        onFilter: (value, todo) => togo.tags.includes(value)
```

> Function description from the docs
<img width="720" alt="image" src="https://user-images.githubusercontent.com/65303812/233571846-3f913354-c54e-4eaa-9d08-f41cad69b45b.png">

Reply:
Okay. I read your comments.  So for question why I used` for loops`. See my way of  writing code mostly take me back to basics. In the moment while writing the code, I wrote what naturally came to me.  I should have refactored though. 

I didn't realise that onFilter returns boolean which it should because filter does that too. Not sure why the code is still working. 

on onFilter, I will apply the code the way you have done it.  It is easier to understand.